### PR TITLE
Update performance test and README to include more fair comparison with pubchempy

### PR DESCRIPTION
**Pull request by jonwzheng**
This PR makes two small changes -
1. Updates the compound list in `test_performance.py` to include only 100 molecules that are known to be translateable using both OPSIN and PubChem. The previous list of compounds had many species that couldn't be resolved using PubChem, resulting in a potentially unfair comparison between the two tools.  Also, the compound list was fairly long - ideally we want to keep the test list as short as possible so as to not spam the PubChem server.
 
2. Updates the README to reflect this change, and modifies the language describing the comparison. In particular, **the speedup will depend on the size of the input.** `py2opsin` has a non-negligible startup time, and if you are only translating 1 species at a time, `py2opsin` performance is actually comparable to querying PubChem. The real speedup comes from the batch translations, which is now indicated in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,19 +53,19 @@ Arguments:
  - jar_fpath (str, optional): Filepath to OPSIN jar file. Defaults to "opsin-cli.jar" which is distributed with py2opsin.
 
 
-## Speedup 50x from `pubchempy`
-`py2opsin` runs locally and is smaller in scope in what it provides, which makes it __dramatically__ faster at resolving identifiers. In the code block below, the call to `py2opsin` will execute ~58x faster than an equivalent call to `puchempy`:
+## Massive speedup from `pubchempy` for batch translations
+`py2opsin` runs locally and is smaller in scope in what it provides, which makes it __dramatically__ faster at resolving identifiers. In the code block below, the call to `py2opsin` will execute ~23 faster than an equivalent call to `pubchempy`:
```
Comment (JacksonBurns) [inline README.md]: Let's get rid of the numerical comparison here - going to vary based on chemicals, machines, internet speed, etc.

File: test/test_performance.py [hunk 2]
```diff
@@ -19,376 +19,109 @@ def setUpClass(self):
             os.getcwd(),
             "opsin-cli-2.7.0-jar-with-dependencies.jar",
         )
-        # list of 10 carbon molecules taken from wikipedia
-        # https://en.wikipedia.org/wiki/List_of_compounds_with_carbon_number_10
+        # list of molecules taken from IUPAC Dissociation Constants dataset
+        # https://zenodo.org/record/7236453
```
Comment (JacksonBurns) [inline test/test_performance.py:23]: Thanks for including this link as well.

**Review by JacksonBurns — CHANGES_REQUESTED** [review]
Just one small note - otherwise I think it's about ready to go! Will need to be updated as well.

File: test/test_performance.py (hunk 2 above)
Comment (jonwzheng) [reply test/test_performance.py:23]: :D

File: README.md (hunk 1 above)
Comment (jonwzheng) [reply README.md]: OK, fixed that up in the latest commits

**Comment by jonwzheng** [comment]
It looks like the performance tests have the possibility of failing if the PubChem server can't be reached. We may want to include error handling 
This might happen if too many queries are sent in a short time (PubChem recommends limiting access to 5 queries per second)... but making that limitation might contaminate the speed comparison.

**Comment by JacksonBurns** [comment]
I think this might actually be a limitation of the GitHub runners - they are notoriously weird. On the second run, it passed.

**Comment by jonwzheng** [comment]
Ah, I spoke too soon - great!

**Review by JacksonBurns — APPROVED** [review]
Thanks for taking care of this one! We will bear in mind the comments made about possible network issues in case this comes up again in the future.

**Pull request merged**